src/ShallowTraversal.js

```diff
@@ -111,6 +111,10 @@ export function buildPredicate(selector) {
 
         case SELECTOR.PROP_TYPE: {
           const propKey = selector.split(/\[([a-zA-Z-0-9-]*?)(=|])/)[1];
+          const pattern = /^[a-zA-Z][a-zA-Z-0-9-]*?/;
```

Comment:
it might be simpler to change the `propKey` splitting regex to be `/\[([a-z]+[a-z-0-9-]*?)(=|])/i`, and do a bit more checking on the split result?